modal/app.py

```diff
@@ -547,6 +550,35 @@ def wrapped(
                 if webhook_config and interactive:
                     raise InvalidError("interactive=True is not supported with web endpoint functions")
             else:
+                if not is_global_object(f.__qualname__) and not serialized:
+                    raise InvalidError(
+                        dedent(
+                            """
+                            The `@app.function` decorator must apply to functions in global scope,
+                            unless `serialize=True` is set.
+                            If trying to apply additional decorators, they may need to use `functools.wraps`.
+                            """
```

Comment:
Does this this error obviate the one we have here: https://github.com/modal-labs/modal-client/blob/ded0d09f61d32f7a6cada0984757975a968bcf10/modal/_utils/function_utils.py#L110-L114?

Reply:
No, the other one is for situations where the a function is defined in local scope, i.e. inside another function.

The one we're adding is for global scope but not top-level scope.